### PR TITLE
Feature/trello 114 refactor

### PR DESCRIPTION
**Pull request by mdvanes**
Please note:

* After consulting @Tenchi2xh, I have opted to disable `LandscapeBackground` in `Intro`, because it uses `SnapSVG` which causes the page bottom margin bug. @Tenchi2xh will look into a solution without `SnapSVG`
* It would be great if someone could look at the `any` types. Maybe @Bjeaurn  or @mvdam 
* `AsyncComponent` has been refactored to Suspense + Lazy, thanks @MikelArnaiz for the suggestion. However, the `FullHeight` component does not seem to work completely. When initially switching tabs, there is still a visible resizing

### CODE REVIEW

File: src/AsyncComponent/AsyncComponent.tsx [hunk 1]
```diff
@@ -1,59 +1,59 @@
...
-  }
-}
-
-const FullHeight: SFC<{}> = props => {
-  const styles: CSSProperties = {
-    minHeight: '100vh',
-  };
-  return <div style={styles}>{props.children}</div>;
-};
+// import React, { Component, FC, CSSProperties, ComponentType } from 'react';
```
Comment (Tenchi2xh) [inline src/AsyncComponent/AsyncComponent.tsx]: Why keep the file?

File: src/Intro/Intro.tsx [hunk 2]
```diff
@@ -46,70 +50,62 @@ const styles: any = (theme: any) => ({
...
 
     return (
       <div>
-        <section id="intro" className={props.classes.section}>
+        <section id="intro" className={classes.section}>
           <Container fullHeightMinusNavBar center marginTopNavBar>
-            <LandscapeBackground className={props.classes.fullVideo} />
+            {/* TODO SnapSVG has performance issues in Chrome and it creates a bottom margin. SnapSVG is no longer
+            under maintenance (no releases for years). Replacing by Greensock might be a good plan
+            <LandscapeBackground className={classes.fullVideo} /> */}
```
Comment (Tenchi2xh) [inline src/Intro/Intro.tsx]: 😭 
I wanna fix this soon, wish I had more time

**Comment by Tenchi2xh** [comment]
I think the other guys will be more useful than me for reviewing technical FE stuff :)

File: src/AsyncComponent/AsyncComponent.tsx (hunk 1 above)
Comment (mdvanes) [reply src/AsyncComponent/AsyncComponent.tsx]: For decoration? No I just forgot, I removed it.

File: src/Intro/Intro.tsx [hunk 3]
```diff
@@ -46,70 +50,62 @@ const styles: any = (theme: any) => ({
...
-function withStylesTyped(myStyles: any) {
-  return (Component: any) => {
-    return withStyles(myStyles)(Component) as any;
-  };
-}
+// FIXME: any type
+const intersperse = (arr: JSX.Element[], sep: string) =>
+  arr
+    .reduce((acc: JSX.Element[], prev: any) => [...acc, prev, sep], [])
+    .slice(0, -1);
```
Comment (Bjeaurn) [inline src/Intro/Intro.tsx]: Why slice the last one?

File: src/Intro/Intro.tsx [hunk 4]
```diff
@@ -46,70 +50,62 @@ const styles: any = (theme: any) => ({
...
 
-function withStylesTyped(myStyles: any) {
-  return (Component: any) => {
-    return withStyles(myStyles)(Component) as any;
-  };
-}
+// FIXME: any type
+const intersperse = (arr: JSX.Element[], sep: string) =>
+  arr
+    .reduce((acc: JSX.Element[], prev: any) => [...acc, prev, sep], [])
```
Comment (Bjeaurn) [inline src/Intro/Intro.tsx]: Shouldn’t `prev` be a `JSX.Element` here?

File: src/Intro/Intro.tsx (hunk 3 above)
Comment (mdvanes) [reply src/Intro/Intro.tsx]: We would have to ask @Tenchi2xh I just moved this function

File: src/Intro/Intro.tsx (hunk 4 above)
Comment (mdvanes) [reply src/Intro/Intro.tsx]: The problem seems to be that `sep` is added to the array too, which is of type string (I assumed). I can't find a typing that works, that was indeed one of my questions :)

File: src/components/Atoms/LoadingMessage/LoadingMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+const LoadingMessage = () => <div>"Loading..."</div>;
```
Comment (MikelArnaiz) [inline src/components/Atoms/LoadingMessage/LoadingMessage.tsx]:
```suggestion
const LoadingMessage = () => <div>Loading...</div>;
```

File: src/__snapshots__/App.test.tsx.snap [hunk 6]
```diff
@@ -25,46 +25,9 @@ Array [
...
-        Object {
-          "isExact": true,
-          "params": Object {},
-          "path": "/",
-          "url": "/",
-        }
-      }
-    />
+    <div>
+      "Loading..."
```
Comment (MikelArnaiz) [inline src/__snapshots__/App.test.tsx.snap]:
```suggestion
      Loading...
```

**Pull request merged**